src/Service.Tests/CosmosTests/MutationTests.cs

```diff
@@ -390,6 +399,245 @@ public async Task UpdateMutationWithOnlyTypenameInSelectionSet()
             Assert.AreEqual(expected, actual);
         }
 
+        /// <summary>
+        /// For mutation operations, both the respective operation(create/update/delete) + read permissions are needed to receive a valid response.
+        /// In this test, Anonymous role is configured with only create permission.
+        /// So, a create mutation executed in the context of Anonymous role is expected to result in
+        /// 1) Creation of a new item in the database
+        /// 2) An error response containing the error message : "The mutation operation {operation_name} was successful but the current user is unauthorized to view the response due to lack of read permissions"
+        ///
+        /// A create mutation operation in the context of Anonymous role is executed and the expected error message is validated.
+        /// Authenticated role has read permission configured. A pk query is executed in the context of Authenticated role to validate that a new
+        /// record was created in the database.
+        /// </summary>
+        [TestMethod]
+        public async Task ValidateErrorMessageForMutationWithoutReadPermission()
+        {
+            const string SCHEMA = @"
+type Planet @model {
```

Comment:
please add value to `@model` directive per https://learn.microsoft.com/azure/data-api-builder/database-specific-features#user-provided-graphql-schema